errors/error_test.go

```diff
@@ -198,3 +198,19 @@ func TestGetRootError(t *testing.T) {
 		}
 	}
 }
+
+func TestFieldsAdd(t *testing.T) {
```

Comment:
What do you think about this one:
```
fields := Fields{"key1": "val1", "key2": "val2"}
extFields := fields.Add("key3", "val3")
expectedFields := Fields{"key1": "val1", "key2": "val2", "key3":"val3"}
assert.Equal(t, expectedFields, extFields)
```
It seems to be more readable and it checks whether the fields were extended with the key without changing/removing the original ones.